test/test_restraint_base.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import print_function, division
+import IMP
+import IMP.algebra
+import IMP.core
+import IMP.pmi
+import IMP.pmi.restraints
+import IMP.test
+
+
+class RestraintBaseTestCase(IMP.test.TestCase):
+
+    def test_functions(self):
+
+        class DistanceRestraint(IMP.pmi.restraints.RestraintBase):
+            def __init__(self, p1, p2, d, k):
+                m = p1.get_model()
+                super(DistanceRestraint, self).__init__(m)
+                f = IMP.core.Harmonic(d, k)
+                s = IMP.core.DistancePairScore(f)
+                r = IMP.core.PairRestraint(self.m, s, (p1, p2))
+                self.rs.add_restraint(r)
+
+        m = IMP.Model()
+        p1 = IMP.Particle(m)
+        IMP.core.XYZ.setup_particle(p1)
+        p2 = IMP.Particle(m)
+        IMP.core.XYZ.setup_particle(p2)
+        r = DistanceRestraint(p1, p2, 0., 1.)
+        self.assertEquals(r.evaluate(), 0.0)
+        output = r.get_output()
+        self.assertEquals(output["DistanceRestraint_Score"], str(0.0))
```

Comment:
Use `assertEqual` rather than `assertEquals`. The latter according to the Python docs is `# The plurals are undocumented.  Keep them that way to discourage use.`